File: apps/web/src/lib/api/skills.ts
import { apiFetch } from "./client";
import type { DocumentType } from "./documents";

export type SkillRecord = {
  id: string;
  name: string;
  description: string;
  version: string;
  skill_type: string;
  supported_document_types: DocumentType[];
  result_schema_path: string;
  status: string;
  source_snapshot: {
    source_type: string;
    source_uri: string | null;
    source_entrypoint: string | null;
    source_revision: string | null;
    source_fingerprint: string | null;
    source_metadata: Record<string, unknown>;
  };
};

export async function listSkills(): Promise<{ skills: SkillRecord[] }> {
  return apiFetch<{ skills: SkillRecord[] }>("/skills");
}

File: apps/api/app/schemas/skills.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.enums import DocumentType, EntityStatus, SelectableDocumentType, SkillSourceType, SkillType


class SkillSourceSnapshot(BaseModel):
    source_type: SkillSourceType
    source_uri: str | None
    source_entrypoint: str | None
    source_revision: str | None
    source_fingerprint: str | None
    source_metadata: dict


class SkillRead(BaseModel):
    id: UUID
    name: str
    description: str
    version: str
    skill_type: SkillType
    supported_document_types: list[DocumentType]
    result_schema_path: str
    status: EntityStatus
    source_snapshot: SkillSourceSnapshot
    created_at: datetime
    updated_at: datetime


class SkillsListResponse(BaseModel):
    skills: list[SkillRead]


class SkillCreate(BaseModel):
    name: str
    description: str
    version: str
    skill_type: SkillType
    supported_document_types: list[SelectableDocumentType] = Field(default_factory=list)
    source_type: SkillSourceType
    source_uri: str | None = None
    source_entrypoint: str | None = None
    source_metadata: dict = Field(default_factory=dict)
    prompt_text: str
    result_schema_path: str


class SkillPatch(BaseModel):
    description: str | None = None
    supported_document_types: list[SelectableDocumentType] | None = None
    source_uri: str | None = None
    source_entrypoint: str | None = None
    source_metadata: dict | None = None
    prompt_text: str | None = None
    result_schema_path: str | None = None
